tests/integration/agents/kyma/test_kyma.py

```diff
@@ -0,0 +1,468 @@
+from textwrap import dedent
+from unittest.mock import Mock
+
+import pytest
+from deepeval import evaluate
+from deepeval.metrics import (
+    FaithfulnessMetric,
+    GEval,
+)
+from deepeval.test_case import LLMTestCase, LLMTestCaseParams
+from langchain_core.messages import AIMessage, HumanMessage, ToolMessage
+
+from agents.common.constants import PLANNER
+from agents.common.state import SubTask
+from agents.kyma.agent import KymaAgent
+from agents.kyma.state import KymaAgentState
+from agents.supervisor.agent import SUPERVISOR
+from integration.agents.fixtures.api_rule import (
+    API_RULE_WITH_WRONG_ACCESS_STRATEGY,
+    EXPECTED_API_RULE_RESPONSE,
+    EXPECTED_API_RULE_TOOL_CALL_RESPONSE,
+    KYMADOC_FOR_API_RULE_VALIDATION_ERROR,
+)
+from integration.agents.fixtures.btp_manager import (
+    EXPECTED_BTP_MANAGER_RESPONSE,
+    RETRIEVAL_CONTEXT,
+)
+from integration.agents.fixtures.serverless_function import (
+    EXPECTED_SERVERLESS_FUNCTION_RESPONSE,
+    EXPECTED_SERVERLESS_FUNCTION_RESPONSE_NO_REPLICAS,
+    FUNCTION_NO_REPLICAS,
+    KYMADOC_FOR_SERVERLESS_FUNCTION_POD_NOT_READY,
+    KYMADOC_FUNCTION_TROUBLESHOOTING,
+    SERVERLESS_FUNCTION_WITH_SYNTAX_ERROR,
+)
+from services.k8s import IK8sClient
+from utils.settings import DEEPEVAL_TESTCASE_VERBOSE
+
+
+@pytest.fixture
+def correctness_metric(evaluator_model):
+    return GEval(
+        name="Correctness",
+        evaluation_params=[
+            LLMTestCaseParams.ACTUAL_OUTPUT,
+            LLMTestCaseParams.EXPECTED_OUTPUT,
+        ],
+        evaluation_steps=[
+            "Evaluate whether two answers are semantically similar or convey the same meaning."
+            "Check whether the facts in 'actual output' contradict any facts in 'expected output'",
+            "Lightly penalize omissions of detail, focusing on the main idea",
+            "Vague language are permissible",
+        ],
+        model=evaluator_model,
+        threshold=0.7,
+    )
+
+
+@pytest.fixture
+def faithfulness_metric(evaluator_model):
+    return FaithfulnessMetric(
+        threshold=0.7,
+        model=evaluator_model,
+        include_reason=True,
+        verbose_mode=DEEPEVAL_TESTCASE_VERBOSE,
+    )
+
+
+@pytest.fixture
+def mock_k8s_client():
+    return Mock(spec_set=IK8sClient)
+
+
+@pytest.fixture
+def kyma_agent(app_models, mock_hana_db_connection_setup):
+    return KymaAgent(app_models)
+
+
+@pytest.mark.parametrize(
+    "test_case,state,context,retrieval_context,expected_result,expected_tool_call,should_raise",
+    [
+        (
+            "API Rule with wrong access strategy",
```

Comment:
I've added when and then in the test function and improved the test_case descriptioon. 
I prefer such parameter because it is displayed when you run the tests.